esmvaltool/recipes/recipe_seaice_extents_sh.yml

```diff
@@ -0,0 +1,58 @@
+# ESMValTool 
+# recipe_seaice_extents_sh.yml
+---
+documentation:
+  title: SH sea ice area recipe
+  description: |
+    This is an example recipe, shared in 
+    CMIP7 hackathon 2024, CSIRO Aspendale (Melbourne).
+    Converted a COSIMA cookbook recipe with help 
+    from Anton Steketee
+
+  authors:
+    - chun_felicity
+    - steketee_anton
+    
+  maintainer:
+    - chun_felicity
+
+  realms:
+    - seaIce
+
+  projects:
+    - access-nri
+
+datasets: 
+
+  - {dataset: ACCESS-ESM1-5, activity: CMIP ,project: CMIP6,  grid: gn,
+  exp: historical, ensemble: r1i1p1f1, start_year: 1981, end_year: 2010} 
+
+# the years 306-366 are comparable to 1958 -2018 
+# re-adjust years in trends script (+1652)
+  - {dataset: ACCESS-OM2, activity: OMIP ,project: CMIP6,  grid: gn,
+  exp: omip2, ensemble: r1i1p1f1, start_year: 329, end_year: 358} 
+
+# observations
+  - {dataset: NSIDC-G02202-sh, project: OBS6, tier: 3,
+  type: reanaly, version: 4, start_year: 1981, end_year: 2010} #- 2018
+
+
+diagnostics:
+
+  sea_ice_sh:
+    description: sea ice area and mapping sea ice concentration
+    variables:
+    ## 2 variables - sea ice concentration and cell area to compute sea ice area
+      si_fraction:
+        short_name: siconc
+        mip: SImon
+      area:
+        short_name: areacello
+        mip: Ofx
+    scripts:
+      map_extents:
+        months: [2,9] # months to map (feb, sep)
+        script: seaice_area_extents/seaice_mapextents.py
+      trends:
+        script: seaice_area_extents/seaicearea_trends.py
```

Comment:
As @valeriupredoi already said, it would be nice to use the [preprocessor functions](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/recipe/preprocessor.html) in the recipe instead of re-implementing them in the diagnostic. The regridding does not quite work yet because the observational dataset here does not have bounds for the latitude and longitude coordinate (there is some work in progress on this in https://github.com/SciTools-incubator/iris-esmf-regrid/pull/276), but for the other diagnostic you could use a recipe that looks like this:

```yaml
preprocessors:
  trends_min:
    area_statistics:
      operator: mean
    annual_statistics:
      operator: min

  trends_max:
    annual_statistics:
      operator: max
    area_statistics:
      operator: mean

datasets: 

  - {dataset: ACCESS-ESM1-5, activity: CMIP ,project: CMIP6,  grid: gn,
     exp: historical, ensemble: r1i1p1f1, start_year: 1981, end_year: 2010} 

# the years 306-366 are comparable to 1958 -2018 
# re-adjust years in trends script (+1652)
  - {dataset: ACCESS-OM2, activity: OMIP ,project: CMIP6,  grid: gn,
     exp: omip2, ensemble: r1i1p1f1, start_year: 329, end_year: 358} 

# observations
  - {dataset: NSIDC-G02202-sh, project: OBS6, tier: 3,
     type: reanaly, version: 4, start_year: 1981, end_year: 2010,
     supplementary_variables: [{short_name: areacello, mip: Ofx}]
     } #- 2018


diagnostics:

  sea_ice_extents_sh:
    description: sea ice extents
    variables:
      siconc:
        mip: SImon
    scripts:
      map_extents:
        months: [2,9] # months to map (feb, sep)
        script: seaice_area_extents/seaice_mapextents.py
  sea_ice_trends_sh:
    description: sea ice trends
    variables:
      siconc_min:
        short_name: siconc
        mip: SImon
        preprocessor: trends_min
      siconc_max:
        short_name: siconc
        mip: SImon
        preprocessor: trends_max
    scripts:
      trends:
        script: seaice_area_extents/seaicearea_trends.py
 ```

Once regridding to a boundless target grid is supported with iris-esmf-regrid, you should be able to add a preprocessor function
```
    regrid:
      scheme:
        reference: esmf_regrid.schemes:ESMFBilinear
        use_tgt_mask: false
      target_grid: NSIDC-G02202-sh
```
and I'm working on making that more convenient in https://github.com/ESMValGroup/ESMValCore/pull/2457, so you could do
```
    regrid:
      scheme: linear
      target_grid: NSIDC-G02202-sh
      use_tgt_mask: false
```

Reply:
Yep, i'll have a play, i think initially when I ported it across some preprocessors weren't working on it but maybe it was issues with the data

Reply:
I think I'm finding the data issues that stopped it from working.. this error below even though `areacello` is fixed and wouldn't have time coordinate? And I'm able to load the `ACCESS-ESM1-5` dataset fine with the supplementary variable and can't see the differences in the file structure. Not sure where to look from here?
```
ValueError: File /g/data/fs38/publications/CMIP6/OMIP/CSIRO-COSIMA/ACCESS-OM2/omip2/r1i1p1f1/Ofx/areacello/gn/latest/areacello_Ofx_ACCESS-OM2_omip2_r1i1p1f1_gn.nc datetimes do not match a recognized pattern and time coordinate can not be read from the file
```
Also, from the recipe, this observational data can't seem to register the supplementary variable either, is it missing something to find it? 🤔

Reply:
Are you running the recipe I posted above? That worked for me, so if it does not work for you, I would recommend looking in the debug log file to see what patterns it is using to look for the files and use that information to adjust your `rootpath` and `drs` settings in config user and the matching settings in config-developer.yml.

Reply:
I'm using the default settings

Reply:
I was in a jupyter notebook as I was figuring out a few more steps.. that error on the `areacello` file was on loading like this where `model` is a Dataset object. And it was finding files ok.
```
model_om = model.copy(**om_facets)
model_om.add_supplementary(short_name='areacello', mip='Ofx')
model_om.load() 
```
I could test the preprocessor steps on the other dataset though to edit the recipe so should be able to move things around 😄

Reply:
> ValueError: File /g/data/fs38/publications/CMIP6/OMIP/CSIRO-COSIMA/ACCESS-OM2/omip2/r1i1p1f1/Ofx/areacello/gn/latest/areacello_Ofx_ACCESS-OM2_omip2_r1i1p1f1_gn.nc datetimes do not match a recognized pattern and time coordinate can not be read from the file

That looks like the kind of error you would get if you specify a timerange (`timerange` or `start_year`/`end_year`) for a dataset that does not have time as a dimension.

The following works for me:
```python
dataset = Dataset(
    short_name="siconc",
    mip="SImon",
    project="CMIP6",
    dataset="ACCESS-OM2",
    grid="gn",
    exp="omip2",
    ensemble="r1i1p1f1",
    timerange="0329/0358",
)
dataset.add_supplementary(short_name="areacello", mip="Ofx")
cube = dataset.load()
```
and `dataset.files` prints
```
[LocalFile('/home/bandela/climate_data/CMIP6/OMIP/CSIRO-COSIMA/ACCESS-OM2/omip2/r1i1p1f1/SImon/siconc/gn/v20210616/siconc_SImon_ACCESS-OM2_omip2_r1i1p1f1_gn_030601-036612.nc')]
```
and `dataset.supplementaries[0].files` prints
```
[LocalFile('/home/bandela/climate_data/CMIP6/OMIP/CSIRO-COSIMA/ACCESS-OM2/omip2/r1i1p1f1/Ofx/areacello/gn/v20210616/areacello_Ofx_ACCESS-OM2_omip2_r1i1p1f1_gn.nc')]
```

Similarly, for the observational dataset,
```python
dataset = Dataset(
    short_name="siconc",
    mip="SImon",
    project="OBS6",
    dataset="NSIDC-G02202-sh",
    timerange="1981/2010",
    tier="3",
    type="reanaly",
)
dataset.add_supplementary(short_name="areacello", mip="Ofx")
cube = dataset.load()
```
works (even if there are many warnings about missing bounds, maybe this is something that could be improved in the CMORizer script?). In this case, `dataset.files` prints
```
[LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198101-198112.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198201-198212.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198301-198312.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198401-198412.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198501-198512.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198601-198612.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198701-198712.nc'),
 LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_SImon_siconc_198801-198812.nc'),
...
```
and `dataset.supplementaries[0].files` prints
```
[LocalFile('/home/bandela/climate_data/Tier3/NSIDC-G02202-sh/OBS6_NSIDC-G02202-sh_reanaly_4_Ofx_areacello.nc')]
```
Does that example help? If not, could you please share your complete code and the content of config-user.yml and config-developer.yml so I can try to reproduce your issue?

Reply:
thanks! maybe my notebook got too messy trying to run it, I'll have to come back to this later and have a play with the regridders as well, have to get some other things done 😅

Reply:
I have looked at this again and regridding won't work because of the missing bounds in the observational data(`NSIDC-G2202`), its a 2 dimensional coordinate though so the `guess_bounds` doesn't work. Is there an easy way of adding these instead of trying to build it? And I can add into the CMOR script :)

Reply:
If you know what the bounds are, then adding them in the CMORizer script would be great as this will enable area weighted/conservative regridding. However, linear regridding should work without bounds with the latest release of iris-esmf-regrid (see the changelog for version [0.11](https://github.com/SciTools-incubator/iris-esmf-regrid/blob/main/CHANGELOG.md#added)).

Reply:
Thanks, yes, I think I'll use the ocean fix grid code to create bounds in the CMORizer script, that would be great to have. I'll check what versions I have and test a few regridders to see how they come out. We have our ACCESS workshop next week so would have to be later. Thanks for your help!

Reply:
I'll post some outputs in another comment, target_grid to a dataset seems to be doing something funky for me. I'll resolve this with the updated files